#VRML_SIM R2022a utf8
WorldInfo {
  basicTimeStep 16
  contactProperties [
    ContactProperties {
      material2 "dumbbell"
      coulombFriction [
        0
      ]
    }
  ]
}
Viewpoint {
  orientation -0.29076668110270615 0.6817799540516973 0.6712904225547857 1.1319667354631833
  position -1.2291310788616117 -1.2892495748739428 2.115310561101891
}
TexturedBackground {
}
TexturedBackgroundLight {
}
Floor {
  size 1 1
}
WoodenBox {
  translation 0.3 0.31 0.05
  size 0.1 0.1 0.1
}
WoodenBox {
  translation 0.06 -0.3 0.05
  name "wooden box(1)"
  size 0.1 0.1 0.1
}
WoodenBox {
  translation -0.35 0.11 0.05
  name "wooden box(2)"
  size 0.1 0.1 0.1
}
E-puck {
  translation 0.45973140642010485 2.8344946319198508e-05 -3.855247540149147e-05
  rotation 0.0011235946696868835 0.0012774848475770033 -0.9999985527826939 1.4491738674470465
  controller "epuck_avoid_collision"
}
Solid {
  translation 0.2 0 0.04958907968091047
  rotation 0 1 0 0
  children [
    Shape {
      appearance PBRAppearance {
        baseColorMap ImageTexture {
          url [
            "../textures/red_brick_wall.jpg"
          ]
        }
        metalness 0
      }
      geometry DEF BALL_GEOMETRY Sphere {
        radius 0.05
        subdivision 2
      }
    }
  ]
  boundingObject USE BALL_GEOMETRY
  physics Physics {
  }
}
Solid {
  translation 0.51 0 0.05
  children [
    DEF WALL Shape {
      appearance PBRAppearance {
        baseColor 0 0.333333 1
        roughness 0.5
        metalness 0
      }
      geometry Box {
        size 0.02 1 0.1
      }
    }
  ]
  name "solid(1)"
  boundingObject USE WALL
}
Solid {
  translation 0 0.51 0.05
  rotation 0 0 1 1.5708003061004252
  children [
    USE WALL
  ]
  name "solid(2)"
  boundingObject USE WALL
}
Solid {
  translation 0 -0.51 0.05
  rotation 0 0 1 1.5707996938995747
  children [
    USE WALL
  ]
  name "solid(3)"
  boundingObject USE WALL
}
Solid {
  translation -0.51 0 0.05
  children [
    USE WALL
  ]
  name "solid(4)"
  boundingObject USE WALL
}
DEF Dumbell Solid {
  translation -0.1799999507270521 0.3 0.029215199999999983
  rotation 0 1 0 1.5739355319508335
  children [
    DEF G0 Group {
      children [
        Transform {
          translation 0 0 -0.05
          children [
            Shape {
              appearance PBRAppearance {
                metalness 0
              }
              geometry Sphere {
                radius 0.03
                subdivision 2
              }
            }
          ]
        }
        Transform {
          translation 0 0 0.05
          children [
            Shape {
              appearance PBRAppearance {
                metalness 0
              }
              geometry Sphere {
                radius 0.03
                subdivision 2
              }
            }
          ]
        }
        Shape {
          appearance PBRAppearance {
            metalness 0
          }
          geometry Cylinder {
            height 0.1
            radius 0.01
          }
        }
      ]
    }
  ]
  name "solid(5)"
  contactMaterial "dumbbell"
  boundingObject USE G0
  physics Physics {
    density -1
    mass 2
    centerOfMass [
      0 0 0.01
    ]
  }
}
Robot {
  translation -3.077350162932461e-17 3.1896523458341972e-09 0.03986048964910958
  rotation -1 -7.315595545279603e-10 -6.6068097357821266e-09 9.578263471891908e-08
  children [
    DistanceSensor {
      translation 0.1 -0.035 0
      rotation 0 0 1 -0.3
      children [
        Solid {
          children [
            DEF DS_RIGHT Shape {
              appearance PBRAppearance {
                baseColor 0 0 1
                metalness 0
              }
              geometry Box {
                size 0.01 0.01 0.01
              }
            }
          ]
          boundingObject USE DS_RIGHT
        }
      ]
      name "ds_right"
    }
    DistanceSensor {
      translation 0.1 0.035 0
      rotation 0 0 1 0.3
      children [
        DEF DS Solid {
          children [
            DEF DSLEFT Shape {
              appearance PBRAppearance {
                baseColor 0 0 1
                metalness 0
              }
              geometry Box {
                size 0.01 0.01 0.01
              }
            }
          ]
          boundingObject USE DSLEFT
        }
      ]
      name "ds_left"
    }
    Shape {
      appearance PBRAppearance {
        baseColor 1 0 0
        metalness 0
      }
      geometry DEF robobox Box {
        size 0.2 0.1 0.05
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 6.1973125603986796e-12
        axis 0 1 0
        anchor 0.05 0.06 0
      }
      device [
        RotationalMotor {
          name "wheel1"
        }
      ]
      endPoint DEF WHEEL1 Solid {
        translation 0.05 0.06 0
        rotation 1 2.1690520927017937e-11 -2.1690593961395376e-11 1.5707996938995745
        children [
          DEF WHEEL Shape {
            appearance PBRAppearance {
              baseColor 0 1 0
              metalness 0
            }
            geometry Cylinder {
              height 0.02
              radius 0.04
            }
          }
        ]
        boundingObject USE WHEEL
        physics DEF WHEEL_PH Physics {
        }
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position 7.356996433158888e-12
        axis 0 1 0
        anchor 0.05 -0.06 0
      }
      device [
        RotationalMotor {
          name "wheel2"
        }
      ]
      endPoint DEF WHEEL2 Solid {
        translation 0.05 -0.06 0
        rotation 1 2.5749385051181948e-11 -2.5749487516056105e-11 1.5708003061004252
        children [
          USE WHEEL
        ]
        boundingObject USE WHEEL
        physics USE WHEEL_PH
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position -6.196378015710633e-12
        axis 0 1 0
        anchor -0.05 0.06 0
      }
      device [
        RotationalMotor {
          name "wheel3"
        }
      ]
      endPoint DEF WHEEL3 Solid {
        translation -0.05 0.06 0
        rotation 1 -2.168725003162324e-11 2.1687323054987216e-11 1.5707996938995745
        children [
          USE WHEEL
        ]
        boundingObject USE WHEEL
        physics USE WHEEL_PH
      }
    }
    HingeJoint {
      jointParameters HingeJointParameters {
        position -7.357089329324116e-12
        axis 0 1 0
        anchor -0.05 -0.06 0
      }
      device [
        RotationalMotor {
          name "wheel4"
        }
      ]
      endPoint DEF WHEEL4 Solid {
        translation -0.05 -0.06 0
        rotation 1 -2.574971018646643e-11 2.5749812652634407e-11 1.5708003061004252
        children [
          USE WHEEL
        ]
        boundingObject USE WHEEL
        physics USE WHEEL_PH
      }
    }
  ]
  boundingObject USE robobox
  physics Physics {
  }
  controller "four_wheeled_collision_avoidance"
}
